hyperspy/_signals/eds_tem.py

```diff
@@ -483,3 +526,66 @@ def create_model(self, auto_background=True, auto_add_lines=True,
                             auto_add_lines=auto_add_lines,
                             *args, **kwargs)
         return model
+
+    def _get_dose(self, method, beam_current='auto', real_time='auto',
+                            area='auto'):
+        """
+        Calculates the total electron dose for the zeta factor or cross section
+        methods of quantification.
+
+        Inputgiven by i*t*N, i the current, t the
+        acquisition time, and N the number of electron by unit electric charge.
+
+        Parameters
+        ----------
+        method : Defined as either 'zfactor' or 'cross_section'
+        If 'zfactor', the dose is given by i*t*N,
+        i is the current
+        t is the dwell time
+        N is the number of electrons by unit charge or 1/e.
+        If 'cross_section', the dose per unit is area is returned by i*t*N/A
+        A is the beam area or pixel area in nm^2
+        beam_current: Defined in nA with the symbol i above.
+        real_time: Defined in s with the symbol t above.
+        area: Defined in nm^2 with the symbol A above, is the illumination area
+        of the electron beam. If none is given it is extracted from the scale
+        axes_manager. Therefore we assume the probe is oversampling such that
+        the illumination area can be approximated to the pixel area of the
+        pectrum image.
+
+        Returns
+        --------
+        Dose in electrons (zfactor) or electrons per nm^2 (cross_section).
+        """
+
+        parameters = self.metadata.Acquisition_instrument.TEM
+        if beam_current == 'auto':
+            if 'beam_current' not in self.metadata.Acquisition_instrument.TEM:
+                raise Exception ('It is not possible to carry out EDX \
+quantification_cross_section without a beam_current please set one using \
+self.metadata.Acquisition_instrument.TEM.beam_current and run quantification \
+again.')
+            else:
+                beam_current = parameters.beam_current
+        if real_time == 'auto':
+            real_time = parameters.Detector.EDS.real_time
+            if 'real_time' not in self.metadata.Acquisition_instrument.TEM.Detector.EDS:
+                raise Exception ('Please note that your real time is set to\
+the default value of 0.5s. The function will still run. However, if this is \
+incorrect you should consider changing it using \
+self.metadata.Acquisition_instrument.TEM.Detector.EDS.real_time .')
+        if method == 'cross_section':
+            if area == 'auto':
+                pixel1 = self.axes_manager[0].scale
+                pixel2 = self.axes_manager[1].scale
+            if self.axes_manager[0].scale == 1 or self.axes_manager[1].scale == 1:
+                warnings.warn('Please note your pixel width is set to the \
```

Comment:
Ok so it should be a logging.warning then? Essentially the default step size is 1nm in the spectrum image. However, I can for see a situation where it might arise that the user wants to use the default value. In most scenarios it will act as a warning that they haven't calibrated their spectrum image properly. I don't think it is possible to have no default in this case because of how the spectrum scale is set up?